networkx/algorithms/distance_measures.py

```diff
@@ -342,9 +342,15 @@ def diameter(G, e=None, usebounds=False, weight=None):
     G : NetworkX graph
        A graph
 
-    e : eccentricity dictionary, optional
+    e : eccentricity dictionary, optional (default = None)
```

Comment:
Minor nit:
```suggestion
    e : eccentricity dictionary, optional (default=None)
```
because
```bash
$ git grep '(default=None)' | wc
    157    1143   14023
$ git grep '(default = None)' | wc
     18     186    1822
```

Reply:
For the sake of simplicity I'm just going to remove the `default` specification from the parameter type description altogether (the way I originally had it). The inclusion of `default` is optional (ha) anyways, and is covered by the parameter description. Thanks all!

Reply:
Sounds good, I also prefer to not specify `default=None`, since that's implied by `, optional` :+1:

Reply:
The term `optional` means you have to scroll up to the function signature to figure out what the default is.
By including it in the parameter description there is one source for info about the parameter.

And the punctuation used should really be a colon instead of equals sign `(default: None)`. 

Both of these comments can be ignored. :)